.github/workflows/e2e-conformance-plugin.yaml

```diff
@@ -0,0 +1,67 @@
+name: E2E Test - Conformance Plugin
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+      - release-*
+    paths:
+      - "cli/cmd/plugin/conformance/**"
+      - "!cli/cmd/plugin/conformance/**.md"
+      - ".github/workflows/e2e-conformance-plugin.yaml"
+    tags-ignore:
+      - "**"
+
+jobs:
+  setup-runner:
+    name: Start self-hosted EC2 runner
+    runs-on: ubuntu-latest
+    outputs:
+      ec2-instance-id: ${{ steps.start-ec2-runner.outputs.ec2-instance-id }}
+    steps:
+      - name: Start EC2 runner
+        id: start-ec2-runner
+        shell: bash
+        run: |
+          echo "GITHUB_RUN_ID: ${GITHUB_RUN_ID}"
+          echo "GITHUB_RUN_NUMBER: ${GITHUB_RUN_NUMBER}"
+          echo "GITHUB_RUN_ATTEMPT: ${GITHUB_RUN_ATTEMPT}"
+          INSTANCE_NAME="id-${GITHUB_RUN_ID}-${GITHUB_RUN_NUMBER}-${GITHUB_RUN_ATTEMPT}"
+          echo "INSTANCE_NAME: ${INSTANCE_NAME}"
+          echo ::set-output name=ec2-instance-id::${INSTANCE_NAME}
+
+  e2e-test-conformance-plugin:
+    name: E2E Test - conformance plugin
+    needs: setup-runner # required to start the main job when the runner is ready
+    runs-on: ${{ needs.setup-runner.outputs.ec2-instance-id }} # run the job on the newly created runner
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v1
+
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v3
+        with:
+          go-version: "1.17"
+        id: go
+
+      - name: Run E2E Test - Conformance Plugin
+        run: |
+          make conformance-e2e-test
```

Comment:
These conformance tests can take quite a long time.

Are we at risk of this timing out? Is there a timeout check we can put on this to keep it going for long periods of time (potentially hours)?

Reply:
Hey @jpmcb Thanks for following up.
We are currently running the test for 180 minutes by running below command in `cli/cmd/plugin/conformance/Makefile`.
```
cd e2e-test && ROOT_DIR=$(ROOT_DIR) go test -tags=e2e -timeout 180m
```
I ran the test some more than 10 times and the time it took is around 90 minutes. So I think we are safe from timing out but yeah it’s not the idol step looking to set the -timeout flag for infinite(till the test Pass or fail) time.

Reply:
Sounds great!!!